components/core/src/clp/streaming_archive/reader/File.cpp

```diff
@@ -257,39 +257,32 @@ SubQuery const* File::find_message_matching_query(Query const& query, Message& m
         // Get number of variables in logtype
         auto const& logtype_dictionary_entry = m_archive_logtype_dict->get_entry(logtype_id);
         auto const num_vars = logtype_dictionary_entry.get_num_variables();
+        auto const vars_ix_end{m_variables_ix + num_vars};
+        if (vars_ix_end > m_num_variables) {
+            // Logtypes not in sync with variables, so stop search
+            return nullptr;
+        }
```

Comment:
would this ever happen in a legal archive?

Reply:
This should no longer be possible. It used to be possible when we supported storing encoded files on disk, before moving them into an archive.